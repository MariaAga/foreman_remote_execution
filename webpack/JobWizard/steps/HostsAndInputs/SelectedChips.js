import React from 'react';
import PropTypes from 'prop-types';
import { Chip, ChipGroup, Button } from '@patternfly/react-core';
import { translate as __ } from 'foremanReact/common/I18n';
import { hostMethods } from '../../JobWizardConstants';

const SelectedChip = ({ selected, setSelected, categoryName }) => {
  const deleteItem = itemToRemove => {
    setSelected(oldSelected =>
      oldSelected.filter(item => item !== itemToRemove)
    );
  };
  return (
    <ChipGroup className="hosts-chip-group" categoryName={categoryName}>
      {selected.map(chip => (
        <Chip key={chip} id={chip} onClick={() => deleteItem(chip)}>
          {chip}
        </Chip>
      ))}
    </ChipGroup>
  );
};

export const SelectedChips = ({
  selectedHosts,
  setSelectedHosts,
  selectedHostCollections,
  setSelectedHostCollections,
  selectedHostGroups,
  setSelectedHostGroups,
}) => (
  <div className="selected-chips">
    <SelectedChip
      selected={selectedHosts}
      categoryName={hostMethods.hosts}
      setSelected={setSelectedHosts}
    />
    <SelectedChip
      selected={selectedHostCollections}
      categoryName={hostMethods.hostCollections}
      setSelected={setSelectedHostCollections}
    />

    <SelectedChip
      selected={selectedHostGroups}
      categoryName={hostMethods.hostGroups}
      setSelected={setSelectedHostGroups}
    />
    <Button variant="link">{__('Clear filters')}</Button>
  </div>
);

SelectedChips.propTypes = {
  selectedHosts: PropTypes.array.isRequired,
  setSelectedHosts: PropTypes.func.isRequired,
  selectedHostCollections: PropTypes.array.isRequired,
  setSelectedHostCollections: PropTypes.func.isRequired,
  selectedHostGroups: PropTypes.array.isRequired,
  setSelectedHostGroups: PropTypes.func.isRequired,
};

SelectedChip.propTypes = {
  categoryName: PropTypes.string.isRequired,
  selected: PropTypes.array.isRequired,
  setSelected: PropTypes.func.isRequired,
};
