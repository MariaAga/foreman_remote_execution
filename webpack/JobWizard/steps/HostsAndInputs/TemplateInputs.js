import React from 'react';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';
import { TemplateInputsFields } from '../form/TemplateInputsFields';

export const TemplateInputs = ({ inputs, value, setValue }) => {
  if (inputs.length)
    return (
      <TemplateInputsFields inputs={inputs} value={value} setValue={setValue} />
    );
  return (
    <p className="gray-text">
      {__('There are no available input fields for the selected template.')}
    </p>
  );
};
TemplateInputs.propTypes = {
  inputs: PropTypes.array.isRequired,
  value: PropTypes.object,
  setValue: PropTypes.func.isRequired,
};

TemplateInputs.defaultProps = {
  value: {},
};
