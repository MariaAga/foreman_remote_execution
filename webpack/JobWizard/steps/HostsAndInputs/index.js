import React, { useEffect, useState } from 'react';
import {
  Title,
  Button,
  Form,
  FormGroup,
  InputGroup,
  Text,
  Spinner,
} from '@patternfly/react-core';
import PropTypes from 'prop-types';
import { useSelector, useDispatch } from 'react-redux';
import { FilterIcon } from '@patternfly/react-icons';
import { debounce } from 'lodash';
import { get } from 'foremanReact/redux/API';
import { translate as __ } from 'foremanReact/common/I18n';
import { resetData } from 'foremanReact/components/AutoComplete/AutoCompleteActions';
import {
  selectJobTemplate,
  selectWithKatello,
  selectHostCount,
} from '../../JobWizardSelectors';
import { SelectField } from '../form/SelectField';
import { SelectedChips } from './SelectedChips';
import { TemplateInputs } from './TemplateInputs';
import { HostSelect } from './HostSelect';
import { HostSearch } from './HostSearch';
import { HostPreviewModal } from './HostPreviewModal';
import {
  HOSTS,
  HOST_COLLECTIONS,
  HOST_GROUPS,
  hostMethods,
  hostsController,
  hostQuerySearchID,
  HOSTS_API,
  hostsToPreviewAmount,
} from '../../JobWizardConstants';

const buildQuery = (selected, search) => {
  const { hosts, hostCollections, hostGroups } = selected;
  const hostsSearch = `(name = ${hosts.join(' or name = ')})`;
  const hostCollectionsSearch = `(host_collection = ${hostCollections.join(
    ' or host_collection = '
  )})`;
  const hostGroupsSearch = `(host_group = ${hostGroups.join(
    ' or host_group = '
  )})`;
  return [
    hosts.length ? hostsSearch : false,
    hostCollections.length ? hostCollectionsSearch : false,
    hostGroups.length ? hostGroupsSearch : false,
    search.length ? `(${search})` : false,
  ]
    .filter(Boolean)
    .join(' or ');
};

const HostsAndInputs = ({
  templateValues,
  setTemplateValues,
  selected,
  setSelected,
  hostsSearchQuery,
  setHostsSearchQuery,
}) => {
  const [isLoading, setIsLoading] = useState(false);
  useEffect(() => {
    debounce(() => {
      setIsLoading(true);
      dispatch(
        get({
          key: HOSTS_API,
          url: '/api/hosts',
          params: {
            search: buildQuery(selected, hostsSearchQuery),
            per_page: hostsToPreviewAmount,
          },
          handleSuccess: () => setIsLoading(false),
        })
      );
    }, 700)();
  }, [
    dispatch,
    selected,
    selected.hosts,
    selected.hostCollections,
    selected.hostCollections,
    hostsSearchQuery,
  ]);
  const templateInputs = useSelector(selectJobTemplate).template_inputs || [];
  const withKatello = useSelector(selectWithKatello);
  const hostCount = useSelector(selectHostCount);
  const dispatch = useDispatch();

  const selectedHosts = selected.hosts;
  const setSelectedHosts = newSelected =>
    setSelected(prevSelected => ({
      ...prevSelected,
      hosts: newSelected(prevSelected.hosts),
    }));
  const selectedHostCollections = selected.hostCollections;
  const setSelectedHostCollections = newSelected =>
    setSelected(prevSelected => ({
      ...prevSelected,
      hostCollections: newSelected(prevSelected.hostCollections),
    }));
  const selectedHostGroups = selected.hostGroups;
  const setSelectedHostGroups = newSelected => {
    setSelected(prevSelected => ({
      ...prevSelected,
      hostGroups: newSelected(prevSelected.hostGroups),
    }));
  };

  const clearSearch = () => {
    dispatch(resetData(hostsController, hostQuerySearchID));
    setHostsSearchQuery('');
  };
  const [hostMethod, setHostMethod] = useState(hostMethods.hosts);
  const [hostPreviewOpen, setHostPreviewOpen] = useState(false);
  return (
    <div className="target-hosts-and-inputs">
      <HostPreviewModal
        isOpen={hostPreviewOpen}
        setIsOpen={setHostPreviewOpen}
        searchQuery={buildQuery(selected, hostsSearchQuery)}
      />
      <Title headingLevel="h2" className="wizard-title">
        {__('Target hosts and inputs')}
      </Title>
      <Form>
        <FormGroup fieldId="host_selection">
          <InputGroup>
            <SelectField
              className="target-method-select"
              toggleIcon={<FilterIcon />}
              fieldId="host_methods"
              options={Object.values(hostMethods)}
              setValue={setHostMethod}
              value={hostMethod}
            />
            {hostMethod === hostMethods.searchQuery && (
              <HostSearch
                setValue={setHostsSearchQuery}
                value={hostsSearchQuery}
              />
            )}
            {hostMethod === hostMethods.hosts && (
              <HostSelect
                selectedHosts={selectedHosts}
                setSelectedHosts={setSelectedHosts}
                apiKey={HOSTS}
                url="/api/hosts?per_page=100"
                placeholderText={__('Filter by hosts')}
              />
            )}
            {withKatello && hostMethod === hostMethods.hostCollections && (
              <HostSelect
                selectedHosts={selectedHostCollections}
                setSelectedHosts={setSelectedHostCollections}
                apiKey={HOST_COLLECTIONS}
                url="/katello/api/host_collections?per_page=100"
                placeholderText={__('Filter by host collections')}
              />
            )}
            {hostMethod === hostMethods.hostGroups && (
              <HostSelect
                selectedHosts={selectedHostGroups}
                setSelectedHosts={setSelectedHostGroups}
                apiKey={HOST_GROUPS}
                url="/api/hostgroups?per_page=100"
                placeholderText={__('Filter by host groups')}
              />
            )}
          </InputGroup>
        </FormGroup>
        <SelectedChips
          selectedHosts={selectedHosts}
          setSelectedHosts={setSelectedHosts}
          selectedHostCollections={selectedHostCollections}
          setSelectedHostCollections={setSelectedHostCollections}
          selectedHostGroups={selectedHostGroups}
          setSelectedHostGroups={setSelectedHostGroups}
          hostsSearchQuery={hostsSearchQuery}
          clearSearch={clearSearch}
        />
        <Text>
          {__('Apply to')}{' '}
          <Button
            variant="link"
            isInline
            onClick={() => setHostPreviewOpen(true)}
          >
            {hostCount} {__('hosts')}
          </Button>{' '}
          {isLoading && <Spinner size="sm" />}
        </Text>
        <TemplateInputs
          inputs={templateInputs}
          value={templateValues}
          setValue={setTemplateValues}
        />
      </Form>
    </div>
  );
};

HostsAndInputs.propTypes = {
  templateValues: PropTypes.object.isRequired,
  setTemplateValues: PropTypes.func.isRequired,
  selected: PropTypes.shape({
    hosts: PropTypes.array.isRequired,
    hostCollections: PropTypes.array.isRequired,
    hostGroups: PropTypes.array.isRequired,
  }).isRequired,
  setSelected: PropTypes.func.isRequired,
  hostsSearchQuery: PropTypes.string.isRequired,
  setHostsSearchQuery: PropTypes.func.isRequired,
};

export default HostsAndInputs;
