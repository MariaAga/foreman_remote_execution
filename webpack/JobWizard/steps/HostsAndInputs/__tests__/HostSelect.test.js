import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { render, fireEvent, screen } from '@testing-library/react';
import * as patternfly from '@patternfly/react-core';
import { mount } from '@theforeman/test';
import { HostSelect } from '../HostSelect';
import { hostsStore } from './HostsAndInputs.fixtures';

jest.spyOn(patternfly, 'ChipGroup');
patternfly.ChipGroup.mockImplementation(props => <div props={props} />);

const apiKey = 'HOSTS_KEY';

const mockStore = configureMockStore([]);
describe('HostSelect', () => {
  it('rendering', () => {
    const component = mount(
      <Provider store={hostsStore}>
        <HostSelect
          selectedHosts={['hosts1,host2']}
          setSelectedHosts={jest.fn()}
          apiKey={apiKey}
          url="/hosts"
          placeholderText="Test hosts"
        />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
  it('too many', () => {
    const tooManyHostsStore = mockStore({
      [apiKey]: {
        subtotal: 101,
        results: [{ name: 'host1' }, { name: 'host2' }, { name: 'host3' }],
      },
    });
    const { container } = render(
      <Provider store={tooManyHostsStore}>
        <HostSelect
          selectedHosts={['hosts1,host2']}
          setSelectedHosts={jest.fn()}
          apiKey={apiKey}
          url="/hosts"
          placeholderText="Test hosts"
        />
      </Provider>
    );
    const openSelectbutton = container.querySelector(
      '.pf-c-select__toggle-arrow'
    );
    fireEvent.click(openSelectbutton);
    const tooManyHosts = screen.queryAllByText(
      'You have %s hosts to display. Please refine your search. + 101'
    );
    expect(tooManyHosts).toHaveLength(1);
  });
});
