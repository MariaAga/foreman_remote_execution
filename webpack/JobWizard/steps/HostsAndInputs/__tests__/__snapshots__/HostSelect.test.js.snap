// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HostSelect rendering 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HostSelect
    apiKey="HOSTS_KEY"
    placeholderText="Test hosts"
    selectedHosts={
      Array [
        "hosts1,host2",
      ]
    }
    setSelectedHosts={[MockFunction]}
    url="/hosts"
  >
    <Select
      aria-describedby=""
      aria-invalid={false}
      aria-label=""
      aria-labelledby=""
      chipGroupComponent={null}
      chipGroupProps={
        Object {
          "className": "hidden-chips",
        }
      }
      className="without_select2"
      clearSelectionsAriaLabel="Clear all"
      createText="Create"
      customBadgeText={null}
      customContent={null}
      direction="down"
      favorites={Array []}
      favoritesLabel="Favorites"
      hasInlineFilter={false}
      inlineFilterPlaceholderText={null}
      inputIdPrefix=""
      isCreatable={false}
      isDisabled={false}
      isGrouped={false}
      isInputValuePersisted={false}
      isOpen={false}
      isPlain={false}
      maxHeight="40vh"
      menuAppendTo="inline"
      noResultsFoundText="No results found"
      onClear={[Function]}
      onCreateOption={[Function]}
      onFilter={null}
      onSelect={[Function]}
      onToggle={[Function]}
      onTypeaheadInputChanged={[Function]}
      ouiaSafe={true}
      placeholderText="Test hosts"
      removeSelectionAriaLabel="Remove"
      selections={
        Array [
          "hosts1,host2",
        ]
      }
      toggleAriaLabel="Options menu"
      toggleIcon={
        <Spinner
          size="sm"
        />
      }
      toggleId={null}
      typeAheadAriaLabel=""
      validated="default"
      variant="typeaheadmulti"
      width="450px"
    >
      <GenerateId
        prefix="pf-random-id-"
      >
        <div
          className="pf-c-select without_select2"
          data-ouia-component-id="OUIA-Generated-Select-typeaheadmulti-1"
          data-ouia-component-type="PF4/Select"
          data-ouia-safe={true}
          style={
            Object {
              "width": "450px",
            }
          }
        >
          <SelectToggle
            aria-label="Options menu"
            aria-labelledby=" pf-select-toggle-id-1"
            className=""
            handleTypeaheadKeys={[Function]}
            hasClearButton={false}
            hasFooter={false}
            id="pf-select-toggle-id-1"
            isActive={false}
            isDisabled={false}
            isOpen={false}
            isPlain={false}
            menuRef={
              Object {
                "current": null,
              }
            }
            onClickTypeaheadToggleButton={[Function]}
            onClose={[Function]}
            onEnter={[Function]}
            onToggle={[Function]}
            parentRef={
              Object {
                "current": <div
                  class="pf-c-select without_select2"
                  data-ouia-component-id="OUIA-Generated-Select-typeaheadmulti-1"
                  data-ouia-component-type="PF4/Select"
                  data-ouia-safe="true"
                  style="width: 450px;"
                >
                  <div
                    class="pf-c-select__toggle pf-m-typeahead"
                  >
                    <div
                      class="pf-c-select__toggle-wrapper"
                    >
                      <span
                        class="pf-c-select__toggle-icon"
                      >
                        <span
                          aria-valuetext="Loading..."
                          class="pf-c-spinner pf-m-sm"
                          role="progressbar"
                        >
                          <span
                            class="pf-c-spinner__clipper"
                          />
                          <span
                            class="pf-c-spinner__lead-ball"
                          />
                          <span
                            class="pf-c-spinner__tail-ball"
                          />
                        </span>
                      </span>
                      <div
                        class="pf-c-chip-group hidden-chips"
                        data-ouia-component-type="PF4/ChipGroup"
                        data-ouia-safe="true"
                      >
                        <div
                          class="pf-c-chip-group__main"
                        >
                          
                          <ul
                            aria-label="Chip group category"
                            class="pf-c-chip-group__list"
                            role="list"
                          >
                            <li
                              class="pf-c-chip-group__list-item"
                            >
                              <div
                                class="pf-c-chip"
                                data-ouia-component-id="OUIA-Generated-Chip-1"
                                data-ouia-component-type="PF4/Chip"
                                data-ouia-safe="true"
                              >
                                <span
                                  class="pf-c-chip__text"
                                  id="pf-random-id-2"
                                >
                                  hosts1,host2
                                </span>
                                <button
                                  aria-disabled="false"
                                  aria-label="Remove"
                                  aria-labelledby="remove_pf-random-id-2 pf-random-id-2"
                                  class="pf-c-button pf-m-plain"
                                  data-ouia-component-id="Remove"
                                  data-ouia-component-type="PF4/Button"
                                  data-ouia-safe="true"
                                  id="remove_pf-random-id-2"
                                  type="button"
                                >
                                  <svg
                                    aria-hidden="true"
                                    fill="currentColor"
                                    height="1em"
                                    role="img"
                                    style="vertical-align: -0.125em;"
                                    viewBox="0 0 352 512"
                                    width="1em"
                                  >
                                    <path
                                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                                    />
                                  </svg>
                                </button>
                              </div>
                            </li>
                          </ul>
                        </div>
                      </div>
                      <input
                        aria-invalid="false"
                        aria-label=""
                        autocomplete="off"
                        class="pf-c-form-control pf-c-select__toggle-typeahead"
                        id="pf-select-toggle-id-1-select-multi-typeahead-typeahead"
                        placeholder="Test hosts"
                        type="text"
                        value=""
                      />
                    </div>
                    <button
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-label="Options menu"
                      aria-labelledby=" pf-select-toggle-id-1"
                      class="pf-c-button pf-c-select__toggle-button pf-m-plain"
                      id="pf-select-toggle-id-1"
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="pf-c-select__toggle-arrow"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 320 512"
                        width="1em"
                      >
                        <path
                          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>,
              }
            }
            type="button"
            variant="typeaheadmulti"
          >
            <div
              className="pf-c-select__toggle pf-m-typeahead"
              onClick={[Function]}
              onKeyDown={[Function]}
            >
              <div
                className="pf-c-select__toggle-wrapper"
              >
                <span
                  className="pf-c-select__toggle-icon"
                >
                  <Spinner
                    size="sm"
                  >
                    <span
                      aria-valuetext="Loading..."
                      className="pf-c-spinner pf-m-sm"
                      role="progressbar"
                    >
                      <span
                        className="pf-c-spinner__clipper"
                      />
                      <span
                        className="pf-c-spinner__lead-ball"
                      />
                      <span
                        className="pf-c-spinner__tail-ball"
                      />
                    </span>
                  </Spinner>
                </span>
                <ChipGroup
                  aria-label="Chip group category"
                  categoryName=""
                  className="hidden-chips"
                  closeBtnAriaLabel="Close chip group"
                  collapsedText="\${remaining} more"
                  defaultIsOpen={false}
                  expandedText="Show Less"
                  isClosable={false}
                  numChips={3}
                  onClick={[Function]}
                  onOverflowChipClick={[Function]}
                  tooltipPosition="top"
                >
                  <GenerateId
                    prefix="pf-random-id-"
                  >
                    <div
                      className="pf-c-chip-group hidden-chips"
                      data-ouia-component-type="PF4/ChipGroup"
                      data-ouia-safe={true}
                    >
                      <div
                        className="pf-c-chip-group__main"
                      >
                        <ul
                          aria-label="Chip group category"
                          className="pf-c-chip-group__list"
                          role="list"
                        >
                          <li
                            className="pf-c-chip-group__list-item"
                            key="0"
                          >
                            <Chip
                              className=""
                              closeBtnAriaLabel="Remove"
                              component="div"
                              isOverflowChip={false}
                              isReadOnly={false}
                              key=".$hosts1,host2"
                              onClick={[Function]}
                              tooltipPosition="top"
                            >
                              <GenerateId
                                prefix="pf-random-id-"
                              >
                                <div
                                  className="pf-c-chip"
                                  data-ouia-component-id="OUIA-Generated-Chip-1"
                                  data-ouia-component-type="PF4/Chip"
                                  data-ouia-safe={true}
                                >
                                  <span
                                    className="pf-c-chip__text"
                                    id="pf-random-id-2"
                                  >
                                    hosts1,host2
                                  </span>
                                  <Button
                                    aria-label="Remove"
                                    aria-labelledby="remove_pf-random-id-2 pf-random-id-2"
                                    id="remove_pf-random-id-2"
                                    onClick={[Function]}
                                    ouiaId="Remove"
                                    variant="plain"
                                  >
                                    <button
                                      aria-disabled={false}
                                      aria-label="Remove"
                                      aria-labelledby="remove_pf-random-id-2 pf-random-id-2"
                                      className="pf-c-button pf-m-plain"
                                      data-ouia-component-id="Remove"
                                      data-ouia-component-type="PF4/Button"
                                      data-ouia-safe={true}
                                      disabled={false}
                                      id="remove_pf-random-id-2"
                                      onClick={[Function]}
                                      role={null}
                                      type="button"
                                    >
                                      <TimesIcon
                                        aria-hidden="true"
                                        color="currentColor"
                                        noVerticalAlign={false}
                                        size="sm"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          aria-labelledby={null}
                                          fill="currentColor"
                                          height="1em"
                                          role="img"
                                          style={
                                            Object {
                                              "verticalAlign": "-0.125em",
                                            }
                                          }
                                          viewBox="0 0 352 512"
                                          width="1em"
                                        >
                                          <path
                                            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                                          />
                                        </svg>
                                      </TimesIcon>
                                    </button>
                                  </Button>
                                </div>
                              </GenerateId>
                            </Chip>
                          </li>
                        </ul>
                      </div>
                    </div>
                  </GenerateId>
                </ChipGroup>
                <input
                  aria-activedescendant={null}
                  aria-invalid={false}
                  aria-label=""
                  autoComplete="off"
                  className="pf-c-form-control pf-c-select__toggle-typeahead"
                  disabled={false}
                  id="pf-select-toggle-id-1-select-multi-typeahead-typeahead"
                  onChange={[Function]}
                  onClick={[Function]}
                  placeholder="Test hosts"
                  type="text"
                  value=""
                />
              </div>
              <button
                aria-expanded={false}
                aria-haspopup="listbox"
                aria-label="Options menu"
                aria-labelledby=" pf-select-toggle-id-1"
                className="pf-c-button pf-c-select__toggle-button pf-m-plain"
                disabled={false}
                id="pf-select-toggle-id-1"
                onClick={[Function]}
                tabIndex={-1}
                type="button"
              >
                <CaretDownIcon
                  className="pf-c-select__toggle-arrow"
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    className="pf-c-select__toggle-arrow"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 320 512"
                    width="1em"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                    />
                  </svg>
                </CaretDownIcon>
              </button>
            </div>
          </SelectToggle>
        </div>
      </GenerateId>
    </Select>
  </HostSelect>
</Provider>
`;
